### PR TITLE
[WIP] New package: osmin-1.11.1

### LINKED ISSUE #41476: Package request: Osmin
### Package name

osmin

### Package homepage

https://github.com/janbar/osmin

### Description

Osmin is an offline-compatible OpenStreetMap client. It supports both desktop and mobile devices (e.g. PinePhone).

### Does the requested package meet the package requirements?

System, Compiled

### Is the requested package released?

Yes

### PR DESCRIPTION
**Pull request by Eloitor**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-musl)

closes #41476

### CODE REVIEW

File: srcpkgs/osmin/template [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+create_wrksrc=yes
+build_style=cmake
+configure_args="-DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ ."
+hostmakedepends="qt5-qmake pkg-config qt5-host-tools qt5-remoteobjects git"
+makedepends="marisa-devel libxml2-devel qt5-declarative-devel qt5-3d clang libgomp-devel
+ libomp-devel qt5-declarative-devel qt5-tools-devel
+ qt5-svg-devel   qt5-quickcontrols2-devel qt5-sensors-devel
+ qt5-multimedia-devel qt5-remoteobjects-devel qt5-location-devel"
+short_desc="GPS Navigator On-Road/Off-Road"
+
```
Comment (paper42) [inline srcpkgs/osmin/template:15]:
```suggestion
```

File: srcpkgs/osmin/template [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Template file for 'osmin'
+pkgname=osmin
+version=1.8.7
+revision=1
+_libosmscout_version=1.9.2
+create_wrksrc=yes
+build_style=cmake
+configure_args="-DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ ."
+hostmakedepends="qt5-qmake pkg-config qt5-host-tools qt5-remoteobjects git"
```
Comment (paper42) [inline srcpkgs/osmin/template:9]: why is git required?

File: srcpkgs/osmin/template [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+makedepends="marisa-devel libxml2-devel qt5-declarative-devel qt5-3d clang libgomp-devel
+ libomp-devel qt5-declarative-devel qt5-tools-devel
+ qt5-svg-devel   qt5-quickcontrols2-devel qt5-sensors-devel
```
Comment (paper42) [inline srcpkgs/osmin/template:12]:
```suggestion
 qt5-svg-devel qt5-quickcontrols2-devel qt5-sensors-devel
```

File: srcpkgs/osmin/template [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Template file for 'osmin'
+pkgname=osmin
+version=1.8.7
+revision=1
+_libosmscout_version=1.9.2
+create_wrksrc=yes
+build_style=cmake
+configure_args="-DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ ."
+hostmakedepends="qt5-qmake pkg-config qt5-host-tools qt5-remoteobjects git"
+makedepends="marisa-devel libxml2-devel qt5-declarative-devel qt5-3d clang libgomp-devel
```
Comment (paper42) [inline srcpkgs/osmin/template:10]: clang in makedepends looks a bit suspicious, is it really required here?

File: srcpkgs/osmin/template [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ https://github.com/janbar/libosmscout/archive/refs/tags/${pkgname}-${_libosmscout_version}.tar.gz"
+checksum="
+ 39ffdeb300de23ddebbd8042ccc9fcbc5411258f3b8f104d089842afc1b5a05e
+ c74e61d07c2447e8959ca5a19c1ffd0c5ae61dd880bf614f12df02e9106fa585"
+nopie_files="/usr/bin/osmin"
+
+pre_configure() {
+	mv osmin-${version}/* .
+	rmdir -v backend/libosmscout
+	mv -v libosmscout-${pkgname}-${_libosmscout_version} backend/libosmscout
```
Comment (paper42) [inline srcpkgs/osmin/template:31]:
```suggestion
	rmdir backend/libosmscout
	mv libosmscout-${pkgname}-${_libosmscout_version} backend/libosmscout
```

File: srcpkgs/osmin/template (hunk 4 above)
Comment (Eloitor) [reply srcpkgs/osmin/template]: Without clang the package is a bit bigger but not by much. I'm removing it.

Comment (Eloitor) [reply srcpkgs/osmin/template]: When running it, I get a segmentation fault. I'm going back to use clang...

File: srcpkgs/osmin/template [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'osmin'
+pkgname=osmin
+version=1.8.7
+revision=1
+_libosmscout_version=1.9.2
+archs="~armv6l-musl ~armv7l ~i686" # libomp-devel
```
Comment (Piraty) [inline srcpkgs/osmin/template:6]: how about duplicating the list for clarity? https://github.com/void-linux/void-packages/blob/d21de3397f4b3f9ccf738e3cfa7ab5f48adc127c/srcpkgs/llvm12/template#L64

Comment (Eloitor) [reply srcpkgs/osmin/template:6]: I'm not sure about this. In llvm12 the variable `_libomp_enable` is used afterwards to create subpackages. I think this is not needed here.

Comment (Piraty) [reply srcpkgs/osmin/template:6]: first, your list is incomplete as is (should use starred globs for `armv[67]`.

second, your point of restricting archs is to break early and not fail mid dependency installation step due to missing dependency `libomp-devel` , isn't it?
so it's technically incorrect to mix whitelist and blacklist this way, as using the blacklist like you do here makes it try build for archs that don't support libomp per llvm12's whitelist, for example `ppc`, `mips*`, `armv5*`.

so you either blacklist *all* unsupported archs (which are a lot and adds maintenance burden, so please don't) or duplicate the whitelist here to be explicit in sync with the limitation imposed by llvm12, therefore make it easy to maintain and technically correct

**Review by Piraty — CHANGES_REQUESTED** [review]
fails with
> QQmlApplicationEngine failed to load component
> qrc:/controls2/osmin.qml:25:1: module "QtGraphicalEffects" is not installed

**Comment by Eloitor** [comment]
> fails with
> 
> > QQmlApplicationEngine failed to load component
> > qrc:/controls2/osmin.qml:25:1: module "QtGraphicalEffects" is not installed

Thank you! I can't test it since I have other packages installed with this dependency...

File: srcpkgs/osmin/template (hunk 6 above)
Comment (Eloitor) [reply srcpkgs/osmin/template]: Thank you, I think it is correct now.

**Comment by Piraty** [comment]
> Thank you! I can't test it since I have other packages installed with this dependency...

you can, use a chroot

**Comment by Eloitor** [comment]
I tied using xchroot. First running osmin I got 

```
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

Aborted
```

Then I tried with `xvfb-run osmin` and I get

```
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
Signal handler is enabled.
User locale setting is C
no file found for translations ':/i18n/osmin_ca.qm' (using default).
Checking installed assets...
Found assets version 1.8.7
Resource directory is /root/osmin/resources
Trying to connect to service ...
Connect to service succeeded
Loading online tile providers from  "/root/osmin/resources/online-tile-providers.json"
Loading providers from  "/root/osmin/resources/map-providers.json"
Loading providers from  "/root/osmin/resources/voice-providers.json"
MapManager ctor
Lookup databases
Basemap found and loaded from ' "/root/osmin/resources/world" '...
Lookup voices at /root/osmin/voices
Overlay tile cache: osmscout::TileCache(0x1e276f8)
terminate called after throwing an instance of 'std::runtime_error'
  what():  locale::facet::_S_create_c_locale name not valid
Aborted
```

How should I run the graphical application in a chroot?

**Comment by Piraty** [comment]
maybe my [bubblejail](https://github.com/Piraty/bubblejail/blob/b0837d3c6d0484a6da6388e2f6fbb16c102cbaf8/bubblejail#L181) can help you get graphical stuff working

**Comment by Piraty** [comment]
ping

**Comment by Eloitor** [comment]
I just updated to the latest 1.10.4 release, working for me on glibc-x86_64. @Piraty can you check if it works for you?

File: srcpkgs/osmin/template [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# Template file for 'osmin'
+pkgname=osmin
+version=1.10.4
+revision=1
+_libosmscout_version=1.10.4
```
Comment (Piraty) [inline srcpkgs/osmin/template:5]: judging from https://github.com/janbar/osmin/tree/1.10.4/backend , it uses a commit present in the libosmscout upstream repo (https://github.com/Framstag/libosmscout) and only applies non-functional patches to debug levels...
did you try to package libosmscout?

**Review by Piraty — CHANGES_REQUESTED** [review]

File: srcpkgs/osmin/template [hunk 8]
```diff
@@ -0,0 +1,31 @@
+# Template file for 'osmin'
+pkgname=osmin
+version=1.9.6
+revision=1
+_libosmscout_version="1.9-20230212"
+archs="x86_64* aarch64* ppc64*" # libomp-devel
+create_wrksrc=yes
+build_style=cmake
+configure_args="-DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ ."
```
[the remark this answers is not in the document]
Comment (Eloitor) [reply srcpkgs/osmin/template:8]: I got this error with this line commented:
```
rmdir: removing directory, 'backend/libosmscout'
renamed '../libosmscout-osmin-1.10.4' -> 'backend/libosmscout'
CMake Warning:
  Ignoring extra path from command line:

   "/builddir/osmin-1.10.4/"


CMake Error: The source directory "/builddir/osmin-1.10.4" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
=> ERROR: osmin-1.10.4_1: do_configure: 'CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" cmake ${cmake_args} ${configure_args} ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} ${wrksrc}/${build_wrksrc}' exited with 1
=> ERROR:   in do_configure() at common/build-style/cmake.sh:76
```

**Comment by Piraty** [comment]
ping

**Comment by Eloitor** [comment]
I'm sorry... I do not have the time to keep up with this PR

**Pull request closed without merging**